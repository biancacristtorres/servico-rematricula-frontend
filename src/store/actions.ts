import { ActionTree } from 'vuex';
import { RootState } from '.';
import { AlunoMutationTypes } from './aluno/mutations';
import { StoreNamespaces } from './namespaces';
import { ResumoMutationTypes } from './resumo/mutations';
import { GlobalMutationTypes } from './mutations';
import { GlobalState } from './state';

export enum GlobalActionTypes {
  LIMPAR_STORE = 'LIMPAR_STORE',
  DEFINIR_LINK_PARA_VOLTAR = 'DEFINIR_LINK_PARA_VOLTAR',
  LIMPAR_LINK_PARA_VOLTAR = 'LIMPAR_LINK_PARA_VOLTAR',
}

const actions: ActionTree<GlobalState, RootState> = {

  async [GlobalActionTypes.LIMPAR_STORE]({ commit }) {
    commit(`${ StoreNamespaces.ALUNO }/${ AlunoMutationTypes.LIMPAR_STORE }`);
    commit(`${ StoreNamespaces.RESUMO }/${ ResumoMutationTypes.LIMPAR_STORE }`);
  },

  async [GlobalActionTypes.DEFINIR_LINK_PARA_VOLTAR]({ commit }, link: string) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, link);
  },

  async [GlobalActionTypes.LIMPAR_LINK_PARA_VOLTAR]({ commit }) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, '');
  },
};

export default actions;
