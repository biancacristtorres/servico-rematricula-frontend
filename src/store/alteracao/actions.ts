import { ActionTree } from 'vuex';
import { AlteracaoState } from './state';
import { AlteracaoMutationsTypes } from './mutations';
import store, { RootState, StoreNamespaces } from '@/store';
import { Container } from 'typescript-ioc';
import { DisciplinaDisponivelService } from '@/services/alteracao/DisciplinaDisponivelService';
import { TurmasActionRequest } from './request/TurmasActionRequest';
import { TurmasService } from '@/services/alteracao/TurmasService';
import { Turma } from '@/model/Turma';
import { TurmasMutationRequest } from './request/TurmasMutationRequest';
import { TurmasEquivalentesService } from '@/services/alteracao/TurmasEquivalentesService';
import { TurmasEquivalentesMutationRequest } from './request/TurmasEquivalentesMutationRequest';
import { AlternarTurmaAtualActionRequest } from './request/AlternarTurmaAtualActionRequest';
import { AlternarTurmaAtualRequest } from './request/AlternarTurmaAtualRequest';
import { TurmasEspeciaisService } from '@/services/alteracao/TurmasEspeciaisService';
import { TurmasEspeciaisMutationRequest } from './request/TurmasEspeciaisMutationRequest';
import { ConcluirAlteracaoDeSucessoActionRequest } from './request/ConcluirAlteracaoDeSucessoActionRequest';
import { Filtro } from '@/model/Filtro';
import { MiniGradeMutationTypes } from '../miniGrade/mutation';

export enum AlteracaoActionTypes {
  DISCIPLINAS = 'DISCIPLINAS',
  TURMAS = 'TURMAS',
  TURMAS_EQUIVALENTES = 'TURMAS_EQUIVALENTES',
  ALTERNAR_TURMA_ATUAL = 'ALTERNAR_TURMA_ATUAL',
  ALTERANDO_TURMAS = 'ALTERANDO_TURMAS',
  ALTERACAO_FINALIZADA = 'ALTERACAO_FINALIZADA',
  PREPARANDO_TURMAS = 'PREPARANDO_TURMAS',
  PREPARACAO_FINALIZADA = 'PREPARACAO_FINALIZADA',
  TURMAS_ESPECIAIS = 'TURMAS_ESPECIAIS',
  INFORMAR_ALTERACAO = 'INFORMAR_ALTERACAO',
  LIMPAR_ALTERACAO = 'LIMPAR_ALTERACAO',
  CONCLUIR_ALTERACAO_DE_SUCESSO = 'CONCLUIR_ALTERACAO_DE_SUCESSO',
  ALTERAR_FILTRO = 'ALTERAR_FILTRO',
}

const actions: ActionTree<AlteracaoState, RootState> = {

  async [AlteracaoActionTypes.DISCIPLINAS]({ commit }, codigoDoAluno: number) {

    const service = (Container.get(DisciplinaDisponivelService) as DisciplinaDisponivelService);

    const disciplinas = await service.obter(codigoDoAluno);

    commit(AlteracaoMutationsTypes.SET_DISCIPLINAS, disciplinas);
  },

  async [AlteracaoActionTypes.TURMAS]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasService) as TurmasService);

    const turmas: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasMutationRequest(
      actionRequest.codigoDaDisciplina, turmas,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS, mutationRequest);
  },

  async [AlteracaoActionTypes.TURMAS_EQUIVALENTES]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasEquivalentesService) as TurmasEquivalentesService);

    const equivalentes: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasEquivalentesMutationRequest(
      actionRequest.codigoDaDisciplina, equivalentes,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS_EQUIVALENTES, mutationRequest);
  },

  async [AlteracaoActionTypes.TURMAS_ESPECIAIS]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasEspeciaisService) as TurmasEspeciaisService);

    const especiais: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasEspeciaisMutationRequest(
      actionRequest.codigoDaDisciplina, especiais,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS_ESPECIAIS, mutationRequest);
  },

  async [AlteracaoActionTypes.ALTERNAR_TURMA_ATUAL]({ commit }, actionRequest: AlternarTurmaAtualActionRequest) {
    const mutationRequest = new AlternarTurmaAtualRequest(
      actionRequest.codigoDaDisciplina, actionRequest.codigoUnico,
    );

    commit(AlteracaoMutationsTypes.ALTERNAR_TURMA_ATUAL, mutationRequest);
  },

  async [AlteracaoActionTypes.ALTERANDO_TURMAS]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERANDO, true);
  },

  async [AlteracaoActionTypes.ALTERACAO_FINALIZADA]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERANDO, false);
  },

  async [AlteracaoActionTypes.PREPARANDO_TURMAS]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_PREPARANDO_TURMAS, true);
  },

  async [AlteracaoActionTypes.PREPARACAO_FINALIZADA]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_PREPARANDO_TURMAS, false);
  },

  async [AlteracaoActionTypes.INFORMAR_ALTERACAO]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERACAO);
  },

  async [AlteracaoActionTypes.LIMPAR_ALTERACAO]({ commit }) {
    commit(AlteracaoMutationsTypes.LIMPAR_ALTERACAO);
  },

  async [AlteracaoActionTypes.CONCLUIR_ALTERACAO_DE_SUCESSO](
    { commit },
    actionRequest: ConcluirAlteracaoDeSucessoActionRequest) {
      const mutationRequest = new AlternarTurmaAtualRequest(
        actionRequest.codigoDaDisciplina, actionRequest.codigoUnico, actionRequest.eTurmaAtual,
      );

      commit(AlteracaoMutationsTypes.ALTERNAR_TURMA_ATUAL, mutationRequest);
      commit(AlteracaoMutationsTypes.SET_ALTERACAO);

      store.commit(`${ StoreNamespaces.MINI_GRADE }/${ MiniGradeMutationTypes.SET_MINI_GRADE }`,
        actionRequest.horariosDaMiniGrade);
  },

  async [AlteracaoActionTypes.ALTERAR_FILTRO]({ commit }, novoFiltro: Filtro) {
    commit(AlteracaoMutationsTypes.SET_NOVO_FILTRO, Object.assign({}, novoFiltro));
  },
};

export default actions;
