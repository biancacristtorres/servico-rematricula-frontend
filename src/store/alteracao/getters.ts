import { GetterTree } from 'vuex';
import { AlteracaoState } from './state';
import { RootState } from '@/store';
import { Disciplina } from '@/model/Disciplina';
import {
  FiltrarStateDeDisciplinaService,
} from '@/services/alteracao/FiltrarStateDeDisciplinaService';

export enum AlteracaoGetterTypes {
  DISCIPLINA = 'DISCIPLINA',
  POSSUI_TURMAS = 'POSSUI_TURMAS',
  TURMA_ATUAL = 'TURMA_ATUAL',
  ESTA_CARREGANDO = 'ESTA_CARREGANDO',
  POSSUI_ALTERACAO = 'POSSUI_ALTERACAO',
  FILTRO = 'FILTRO',
  TURMAS_FILTRADAS = 'TURMAS_FILTRADAS',
  TURMAS_EQUIVALENTES_FILTRADAS = 'TURMAS_EQUIVALENTES_FILTRADAS',
  TURMAS_ESPECIAIS_FILTRADAS = 'TURMAS_ESPECIAIS_FILTRADAS',
}

const filtrarStateDeDisciplinaService = new FiltrarStateDeDisciplinaService();

const getters: GetterTree<AlteracaoState, RootState> = {

  [AlteracaoGetterTypes.ESTA_CARREGANDO](state) {
    return state.alterando || state.preparandoTurmas;
  },

  [AlteracaoGetterTypes.DISCIPLINA](state) {
    const encontrarDisciplina = (codigoDaDisciplina: number) => filtrarStateDeDisciplinaService
      .processar(state.disciplinas, codigoDaDisciplina);

    return encontrarDisciplina;
  },

  [AlteracaoGetterTypes.POSSUI_TURMAS](state) {

    const possuiTurmas = (codigoDaDisciplina: number) => {
      const disciplina: Disciplina = filtrarStateDeDisciplinaService
        .processar(state.disciplinas, codigoDaDisciplina);

      return disciplina.turmas.length > 0 ||
        disciplina.turmasEquivalentes.length > 0 ||
        disciplina.turmasEspeciais.length > 0;
    };

    return possuiTurmas;
  },

  [AlteracaoGetterTypes.TURMA_ATUAL](state) {

    const turmaAtual = (codigoDaDisciplina: number) => {
      const disciplina: Disciplina = filtrarStateDeDisciplinaService
        .processar(state.disciplinas, codigoDaDisciplina);

      const atualDasTurmas = disciplina.turmas.find((t) => t.eTurmaAtual);
      const atualDasEquivalentes = disciplina.turmasEquivalentes.find((t) => t.eTurmaAtual);
      const atualDasEspeciais = disciplina.turmasEspeciais.find((t) => t.eTurmaAtual);

      return atualDasTurmas || atualDasEquivalentes || atualDasEspeciais;
    };

    return turmaAtual;
  },

  [AlteracaoGetterTypes.POSSUI_ALTERACAO](state) {
    return state.possuiAlteracao;
  },

  [AlteracaoGetterTypes.FILTRO](state) {
    return state.filtro;
  },

  [AlteracaoGetterTypes.TURMAS_FILTRADAS](state) {
    const turmaFiltrada = (codigoDaDisciplina: number) => {
      const disciplina: Disciplina = filtrarStateDeDisciplinaService.processar(
        state.disciplinas, codigoDaDisciplina,
      );

      return state.filtro.exibirTurnosDiferentes ?
        disciplina.turmas :
        disciplina.turmas.filter((t) => t.eDoTurnoDoAluno);
    };

    return turmaFiltrada;
  },

  [AlteracaoGetterTypes.TURMAS_EQUIVALENTES_FILTRADAS](state) {
    const turmasEquivalentesFiltradas = (codigoDaDisciplina: number) => {
      const disciplina: Disciplina = filtrarStateDeDisciplinaService.processar(
        state.disciplinas, codigoDaDisciplina,
      );

      return state.filtro.exibirTurnosDiferentes ?
        disciplina.turmasEquivalentes :
        disciplina.turmasEquivalentes.filter((t) => t.eDoTurnoDoAluno);
    };

    return turmasEquivalentesFiltradas;
  },

  [AlteracaoGetterTypes.TURMAS_ESPECIAIS_FILTRADAS](state) {
    const turmasEspeciaisFiltradas = (codigoDaDisciplina: number) => {
      const disciplina: Disciplina = filtrarStateDeDisciplinaService.processar(
        state.disciplinas, codigoDaDisciplina,
      );

      return state.filtro.exibirTurnosDiferentes ?
        disciplina.turmasEspeciais :
        disciplina.turmasEspeciais.filter((t) => t.eDoTurnoDoAluno);
    };

    return turmasEspeciaisFiltradas;
  },

};

export default getters;
