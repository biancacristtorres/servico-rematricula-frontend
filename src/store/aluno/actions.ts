import { ActionTree } from 'vuex';
import { AlunoState } from './state';
import { RootState } from '@/store';
import { AlunoMutationTypes } from './mutations';
import { Container } from 'typescript-ioc';
import { AlunoService } from '@/services/AlunoService';

export enum AlunoActionTypes {
  OBTER_INFORMACOES_ALUNO = 'OBTER_INFORMACOES_ALUNO',
  LIMPAR_INFORMACOES_ALUNO = 'LIMPAR_INFORMACOES_ALUNO',
  OBTENDO_INFORMACOES_ALUNO = 'OBTENDO_INFORMACOES_ALUNO',
  INFORMACOES_ALUNO_OBTIDAS = 'INFORMACOES_ALUNO_OBTIDAS',
}

const actions: ActionTree<AlunoState, RootState> = {

  async [AlunoActionTypes.OBTER_INFORMACOES_ALUNO]({ commit }, codigoDoAluno: number) {
    const service = (Container.get(AlunoService) as AlunoService);
    const informacoesAluno = await service.obterInformacoesAluno(codigoDoAluno);
    commit(AlunoMutationTypes.SET_INFORMACOES_ALUNO, informacoesAluno);
  },

  async [AlunoActionTypes.LIMPAR_INFORMACOES_ALUNO]({ commit }) {
    commit(AlunoMutationTypes.LIMPAR_STORE);
  },

  async [AlunoActionTypes.OBTENDO_INFORMACOES_ALUNO]({ commit }) {
    commit(AlunoMutationTypes.SET_OBTENDO_INFORMACOES_ALUNO, true);
  },

  async [AlunoActionTypes.INFORMACOES_ALUNO_OBTIDAS]({ commit }) {
    commit(AlunoMutationTypes.SET_OBTENDO_INFORMACOES_ALUNO, false);
  },

};

export default actions;
