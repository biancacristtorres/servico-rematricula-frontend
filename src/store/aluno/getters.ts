import { GetterTree } from 'vuex';
import { AlunoState } from './state';
import { RootState } from '@/store';

export enum AlunoGetterTypes {
  INFORMACOES_ALUNO = 'INFORMACOES_ALUNO',
  ESTA_OBTENDO_INFORMACOES_ALUNO = 'ESTA_OBTENDO_INFORMACOES_ALUNO',
}

const getters: GetterTree<AlunoState, RootState> = {

  [AlunoGetterTypes.INFORMACOES_ALUNO](state) {
    return state.informacoesAluno;
  },

  [AlunoGetterTypes.ESTA_OBTENDO_INFORMACOES_ALUNO](state) {
    return state.estaObtendoInformacoes;
  },

};

export default getters;
