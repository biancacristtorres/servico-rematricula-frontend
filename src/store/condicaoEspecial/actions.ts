import { ActionTree } from 'vuex';
import { CondicaoEspecialState } from './state';
import { RootState } from '@/store';
import { MutationsTypes } from './mutations';
import { Container, AutoWired } from 'typescript-ioc';
import { CondicaoEspecialService } from '@/services/CondicaoEspecialService';
import { CondicaoEspecialRequest } from './request/CondicaoEspecialRequest';

export enum ActionTypes {
  PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS = 'PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS',
  SOLICITAR_CONDICOES_ESPECIAIS = 'SOLICITAR_CONDICOES_ESPECIAIS',
}

const actions: ActionTree<CondicaoEspecialState, RootState> = {

  async [ActionTypes.PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS]({ commit, getters }, codigoDoAluno: number) {
    const service = (Container.get(CondicaoEspecialService) as CondicaoEspecialService);
    const solicitadas = await service.solicitadasPor(codigoDoAluno);

    commit(MutationsTypes.SET_SOLICITADAS, solicitadas);
  },

  async [ActionTypes.SOLICITAR_CONDICOES_ESPECIAIS]({ commit }, request: CondicaoEspecialRequest) {
    const service = (Container.get(CondicaoEspecialService) as CondicaoEspecialService);
    await service.solicitar(request.codigoDoAluno, request.condicoesEspeciais);
  },

};

export default actions;
