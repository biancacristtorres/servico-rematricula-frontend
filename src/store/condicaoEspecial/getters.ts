import { GetterTree } from 'vuex';
import { CondicaoEspecialState } from './state';
import { RootState } from '@/store';
import { CondicaoEspecial } from '@/common/enum/CondicaoEspecial';

export enum GetterTypes {
  POSSUI_SOLICITADAS = 'POSSUI_SOLICITADAS',
  CONDICOES_ESPECIAIS = 'CONDICOES_ESPECIAIS',
}

const getters: GetterTree<CondicaoEspecialState, RootState> = {

  [GetterTypes.CONDICOES_ESPECIAIS](state) {
    return [
      CondicaoEspecial.TurmasEspecias,
      CondicaoEspecial.CampusDiferentes,
      CondicaoEspecial.TurnosDiferentes,
    ];
  },
};

export default getters;
