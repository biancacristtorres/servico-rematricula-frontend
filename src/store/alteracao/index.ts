import { Module } from 'vuex';
import { AlteracaoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const alteracao: Module<AlteracaoState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default alteracao;
