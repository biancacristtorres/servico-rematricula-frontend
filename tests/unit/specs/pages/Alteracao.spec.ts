import { AlteracaoFactory } from '../../factory/pages/AlteracaoFactory';
import store, { StoreNamespaces } from '@/store';
import { AlteracaoGetterTypes } from '@/store/alteracao/getters';
import { AlteracaoActionTypes } from '@/store/alteracao/actions';

describe('Página de Interesse', () => {
  const factory = new AlteracaoFactory();

  beforeEach(() => {
    factory.prepararContainerDeInjecaoDaApi();
    factory.dadoQueApiRetorneStatusDeSucesso();
    factory.espionarApi();
  });

  afterEach(() => {
    factory.restaurarContainerDeInjecaoDaApi();
  });

  it('Ao criar a página de alteração dispara um request de preparação de turmas', async (done) => {
    await factory.montarComponente();

    expect(factory.espiao).toHaveBeenCalledTimes(1);
    done();
  });

  it('Ao criar o componente, o estado de loading é alternado', async (done) => {
    await factory.montarComponente();

    expect(factory.preparandoTurmas).toBeCalled();
    expect(factory.preparacaoFinalizada).toBeCalled();
    done();
  });

  it('O link para voltar de Alterção é o Resumo', async (done) => {
    await factory.montarComponente();

    expect(factory.linkParaVoltar()).toEqual('/Resumo');
    done();
  });

  it('Ao preparar turmas com sucesso, a store de mini grade é carregada', async (done) => {
    await factory.montarComponente();

    expect(factory.stateDaMiniGrade.horarios).toContainEqual(factory.horarioDaMiniGradeEsperado);
    done();
  });

  it('Ao criar componente, limpa as alterações', async (done) => {
    store.dispatch(`${ StoreNamespaces.ALTERACAO }/${ AlteracaoActionTypes.INFORMAR_ALTERACAO }`);

    await factory.montarComponente();

    const atual = store.getters[
      `${ StoreNamespaces.ALTERACAO }/${ AlteracaoGetterTypes.POSSUI_ALTERACAO }`
    ];
    expect(atual).toBeFalsy();
    done();
  });

  it('Quando a feature de filtro está desabilitada, o componente de filtro não é renderizado', async (done) => {
    factory.dadoAToggleDeFiltroDesabilitada();

    await factory.montarComponente();

    const existeFiltroDeTurmas = factory.componente.find('#filtro-de-turmas').exists();
    expect(existeFiltroDeTurmas).toBeFalsy();
    done();
  });

  it('Quando a feature de filtro está habilitada, o componente de filtro é renderizado', async (done) => {
    factory.dadoAToggleDeFiltroHabilitada();

    await factory.montarComponente();

    const existeFiltroDeTurmas = factory.componente.find('#filtro-de-turmas').exists();
    expect(existeFiltroDeTurmas).toBeTruthy();
    done();
  });

  it('snapshot', async (done) => {
    await factory.montarComponente();

    expect(factory.componente).toMatchSnapshot();
    done();
  });

});
