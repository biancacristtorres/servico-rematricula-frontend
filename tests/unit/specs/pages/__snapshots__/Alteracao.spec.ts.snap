// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Página de Interesse snapshot 1`] = `
<vcontainer-stub tag="div" class="pa-0">
  <vlayout-stub tag="div" row="" wrap="" class="pa-0">
    <vflex-stub tag="div" xs12="" md8="" class="px-0 pt-4">
      <vsystembar-stub>
        <vtoolbartitle-stub class="subheading font-weight-bold">
          Disciplinas disponíveis
        </vtoolbartitle-stub>
        <vspacer-stub></vspacer-stub>
        <vtoolbaritems-stub id="filtro-de-turmas" class="hidden-sm-and-down">
          <filtrodeturma-stub></filtrodeturma-stub>
        </vtoolbaritems-stub>
      </vsystembar-stub>
    </vflex-stub>
    <vflex-stub tag="div" xs12="" md8="">
      <disciplinasdisponiveis-stub></disciplinasdisponiveis-stub>
    </vflex-stub>
    <vflex-stub tag="div" xs12="" md4="">
      <vcontainer-stub tag="div" class="px-2">
        <salvaralteracoes-stub></salvaralteracoes-stub>
        <minigradedehorario-stub></minigradedehorario-stub>
      </vcontainer-stub>
    </vflex-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
